account-kit/smart-contracts/src/ma-v2/account/semiModularAccount7702.ts

```diff
@@ -0,0 +1,113 @@
+import type {
+  EntryPointDef,
+  SmartAccountSigner,
+  ToSmartContractAccountParams,
+} from "@aa-sdk/core";
+import {
+  getEntryPoint,
+  toSmartContractAccount,
+  EntityIdOverrideError,
+} from "@aa-sdk/core";
+import { type Chain, type Hex, type Transport, type Address } from "viem";
+import { DEFAULT_OWNER_ENTITY_ID } from "../utils.js";
+import { singleSignerMessageSigner } from "../modules/single-signer-validation/signer.js";
+import { nativeSMASigner } from "./nativeSMASigner.js";
+import {
+  type SignerEntity,
+  type MAV2Account,
+  createMAv2BaseFunctions,
+} from "./common/modularAccountV2Base.js";
+
+export type CreateSMA7702AccountParams<
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+> = Pick<
+  ToSmartContractAccountParams<"MAV2Account", TTransport, Chain, "0.7.0">,
+  "transport" | "chain" | "accountAddress"
+> & {
+  signer: TSigner;
+  entryPoint?: EntryPointDef<"0.7.0", Chain>;
+  signerEntity?: SignerEntity;
```

Comment:
oh yea I remember seeing this during hackathon and thinking it odd. Why can't we merge `signer` and `signerEntity`?

Reply:
cc @howydev